site/docs/v1/tech/lambdas/testing-lambdas.md

```diff
@@ -0,0 +1,380 @@
+- The `createRandomUser` function, which creates a User request object with `const request` and then sends it to `fusion.register()`. Details on this can be found in the [TypeScript client library interface](https://github.com/FusionAuth/fusionauth-typescript-client/blob/master/src/FusionAuthClient.ts).
+- The `deleteUser` function, which you can use in the tests to delete the user just created.
+
+Run the code to test user creation with the following command.
+
+```bash
+node test.js
+```
+
+In FusionAuth, click on <span>Users</span>{:.breadcrumb} to check that a new user called `lambdatestuser` has been created. You can delete the `createRandomUser(uuidv4());` line in `test.js` as each test will use a new temporary user. This will allow you to add multiple lambda tests while avoiding potential conflicts between test users and permissions.
+
+#### Write The Test
+
+Now you will test that the lambda returns "Goodbye World", which will confirm that the CLI `update` command worked.
+
+Add this code to the `test.js` file.
+
+```js
+{% remote_include 'https://raw.githubusercontent.com/FusionAuth/fusionauth-example-testing-lambdas/main/complete-application/documentation_snippets/test_2.js' %}
+```
+
+The `login` function calls the FusionAuth Typescript library. It then decodes the JWT response and returns its `message` property.
+
+This `login` function is called by the `tape` function `test`. This test specifies the name of the test, says that it expects exactly one assertion to occur with `plan`, checks that calling `login` returns the property you expect from the lambda updated earlier, and exits. Even if the test fails, the `finally` clause will delete the temporary user created.
+
+Run it with the following command.
+
+```bash
+node test.js
+```
+
+When your code has several tests, and you want a colorful, concise summary, you can use the following instead.
+
+```bash
+node test.js | npx faucet
+```
+
+### Unit Test: Call An External Service
+
+The next test you'll write is a unit test that verifies your lambda locally using a fake mock service and not in FusionAuth. The benefit of this test is that you can test your logic works without needing an external service to be reliable at the time of testing. The danger is that your test might pass locally, but the lambda might fail on FusionAuth due to it running on a different JavaScript environment with different restrictions and configuration.
+
+Let's take an example where you check if users have email addresses from a country sanctioned by the United States, such as North Korea or Cuba. You call the external site `https://issanctioned.example.com` with an email address, and you're told whether or not the domain is banned.
+
+Add this new function to `test.js`. 
+
+```js
+{% remote_include 'https://raw.githubusercontent.com/FusionAuth/fusionauth-example-testing-lambdas/main/complete-application/documentation_snippets/test_3.js' %}
```

Comment:
I don't understand the point of putting this into a documentation snippet? The point of pulling this from a remote server is that the code will be runnable and stay up to date. See for example the doc here: https://fusionauth.io/docs/v1/tech/themes/client-side-password-rule-validation

How does putting these into a documentation snippet help with this goal?

Reply:
How about instead you build up a test suite that the user can run?